ui/src/geometry/geometry-api.ts

```diff
@@ -273,12 +275,22 @@ export async function getTrackLayoutPlans(
     ).then((plans) => plans.filter(filterNotEmpty));
 }
 
-export async function fetchProjects(): Promise<Project[]> {
-    return await getThrowError<Project[]>(`${GEOMETRY_URI}/projects`);
+export async function getProjects(): Promise<Project[]> {
+    const changeTime = getChangeTimes().project;
+
+    return projectCache.get(changeTime, undefined, () =>
+        getThrowError<Project[]>(`${GEOMETRY_URI}/projects`),
+    );
 }
 
 export async function getProject(id: ProjectId): Promise<Project> {
-    return await getThrowError<Project>(`${GEOMETRY_URI}/projects/${id}`);
+    return getProjects().then((projects) => {
+        const project = projects.find((project) => project.id === id);
```

Comment:
Tätä voisi siis edelleen tehostaa kakuttamalla myös jokaisen yksittäisen projektihakutuloksen. Nythän siis koko projektilistaa käydään mahdollisesti useaan kertaan läpi samalle projektille.

Reply:
Määrä taitaa olla sen verran pieni että ei merkitystä. Ihan ok näin.